### PR TITLE
HOSTEDCP-1104,HOSTEDCP-1105: Validate multi-arch aspects in an AWS Hosted Cluster

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
Validate a multi-arch release image is used when an AWS Hosted Cluster
is set up for multi-arch. Validate management and nodepool CPU arches
match in the CLI if the multi-arch flag is not set. Also, validate each
NodePool CPU arch matches the management cluster's CPU arch when an AWS
Hosted Cluster is not setup for multi-arch.

**Which issue(s) this PR fixes**:
Fixes [HOSTEDCP-1105](https://issues.redhat.com/browse/HOSTEDCP-1105)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: support/util/util.go [hunk 1]
```diff
@@ -318,3 +319,13 @@ func ApplyAWSLoadBalancerSubnetsAnnotation(svc *corev1.Service, hcp *hyperv1.Hos
 		svc.Annotations["service.beta.kubernetes.io/aws-load-balancer-subnets"] = subnets
 	}
 }
+
+func DoesMgmtClusterAndNodePoolCPUArchMatch(nodePoolArch string) error {
```
Comment (jparrill) [inline support/util/util.go:323]: It's there any unit test for this?

**Review by jparrill — COMMENTED** [review]
I think we need a unit test to cover the check for HC + NP images arch. Makes sense?

File: support/util/util.go (hunk 1 above)
Comment (jparrill) [reply support/util/util.go:323]: nah nvm, in order to simulate different arches we will need to do the unit tests for each arch specifically.

**Review by jparrill — COMMENTED** [review]
/lgtm

**Comment by bryan-cox** [comment]
/override "Red Hat Konflux / hypershift-operator-main-on-pull-request"

**Pull request merged**